src/Datasource/ModelAwareTrait.php

```diff
@@ -114,4 +129,22 @@ public function modelFactory($type, callable $factory)
     {
         $this->_modelFactories[$type] = $factory;
     }
+
+    /**
+     * Set or get the model type to be used by this class
+     *
+     * @param string|null $modelType The model type or null to retrieve the current
+     *
+     * @return string|$this
+     */
+    public function modelType($modelType = null)
+    {
+        if ($modelType === null) {
+            return $this->_modelType;
+        }
+
+        $this->_modelType = $modelType;
+
+        return $this;
```

Comment:
@ADmad Most other places the documention just lies: https://github.com/cvo-technologies/cakephp/blob/feature-model-aware-type/src/Datasource/EntityTrait.php#L849
It actually not always returns a string.
Using null or $this as return type or the options, since the first has no benefit, and the 2nd doesn't hurt, I think it is good to be consistent with most of the rest of the framework and use fluent interfaces by design.